aws/rust-runtime/aws-config/src/profile/credentials/exec.rs

```diff
@@ -100,6 +101,9 @@ impl ProviderChain {
                     })?
             }
             BaseProvider::AccessKey(key) => Arc::new(key.clone()),
+            BaseProvider::CredentialProcess(credential_process) => Arc::new(
+                CredentialProcessProvider::new(credential_process.to_string()),
```

Comment:
if `new` returned a result, we could return a profile file error here which is probably the right thing

Reply:
I mentioned this in https://github.com/awslabs/smithy-rs/pull/1356#discussion_r863130251 but I actually ended up pulling out the command parsing logic, following suit with the Java SDK of just passing through the commands to `sh -c` / `cmd.exe /C`.